### ISSUE TITLE
Move `MetricManifold` to ManifoldsBase.jl and extend its interface

### ISSUE THREAD

**Issue by kellertuer** [JuliaManifolds/ManifoldsBase.jl#140]
I think the metric manifold decorator is mature enough to be moved here,
maybe even with a few metric types (maybe also just `EuclideanMetric`).

We could use this also to introduce the function

`get_metric(::AbstracManifold)`

which would return just the `.metric` field for a `MetricManifold` and otherwise the default metric (stored in a trait, so maybe a little tricky to obtain, but surely doable).

**Comment by mateuszbaran** [comment]
Yes, I think basic definitions of `MetricManifold` could be moved here. 

> get_metric(::AbstracManifold)

We already have the `metric` function which does that for `MetricManifold`. I think we could just extend it to other manifolds.

> (stored in a trait, so maybe a little tricky to obtain, but surely doable).

That should be fine if `metric`/`get_metric` becomes a trait function.
